### ISSUE TITLE
Runnable should provide probes?

### ISSUE THREAD

**Issue by prometherion** [kubernetes-sigs/controller-runtime#1183]
I'm using the `controller-runtime` for many purposes, from implementing CRDs reconciliation loop as well for other utilities, many kudos for this amazing project that I'm loving so much.

For a project of mine, I'm implementing a simple HTTP server with some business logic, and thanks to the [`Runnable` interface](https://github.com/kubernetes-sigs/controller-runtime/blob/fdc6658a141b99a3fcb733c8a8000f98e6666f48/pkg/manager/manager.go#L269-L277) I can easily add it to the Manager, as following:

```go
r, _ := webserver.SimpleHttpServerAsExample()
err = mgr.Add(r)
if err != nil {
    log.Error(err, "cannot add Runnable to the Manager")
    os.Exit(1)
}
```

Obviously, besides that, I got several other components running as a Field Indexer, Webhooks, Controllers, and so on and so forth: I expect them to fail at initialization time since they're _safe_ enough since crafted and tested by the community, and (I guess) they're backed by goroutines with some recovery mechanism.

Right now, handling Probes with controller-runtime is something easy to achieve:

```go
import (
	ctrl "sigs.k8s.io/controller-runtime"
	"sigs.k8s.io/controller-runtime/pkg/healthz"
)

func main () {
	mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
		Scheme:                 scheme,
		HealthProbeBindAddress: ":8081",
	})
	_ = mgr.AddReadyzCheck("ping", healthz.Ping)
	_ = mgr.AddHealthzCheck("ping", healthz.Ping)
}
```

But the `healthz.Ping` func is really _dummy_:
```go
var Ping Checker = func(_ *http.Request) error { return nil }
```

But how should I deal with, taking as the said example, a web server? Sure, I cannot use the provided Webhook server, so implementing the Runnable is the most suitable solution but I'd like to achieve also a readiness probe to avoid handling traffic when my instance is not yet ready.

To bypass this problem, I decorated the `Runnable` interface as following:

```go
import (
	"net/http"
	
	"sigs.k8s.io/controller-runtime/pkg/manager"
)

type MyRunnable interface {
	manager.Runnable
	ReadinessProbe(req *http.Request) error
	LivenessProbe(req *http.Request) error
}
```

...so I can do something like this, especially for the living one since the business required:

```go
	var r webserver.MyRunnable
	r, _ = webserver.NewRunnable()
	if err = mgr.AddReadyzCheck("ready", r.ReadinessProbe); err != nil {
		log.Error(err, "cannot create readiness probe")
		os.Exit(1)
	}
```

### tl;dr;

Just throwing idea since my daily use of the package: should we consider extending the `Runnable` interface to provide these probes or should we implement them as I did in the said example?

If we don't make it breaking change, we could think something as implemented with the [`Client` interface](https://github.com/kubernetes-sigs/controller-runtime/blob/fdc6658a141b99a3fcb733c8a8000f98e6666f48/pkg/runtime/inject/inject.go#L73-L77), registering the probes if interface is satisfied.
